### PR TITLE
fix: pthread_mutex_unlock update owner atomically

### PR DESCRIPTION
**Pull request by abhidixi11**
Added critical section protection to update owner and unlock atomically

Description
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

### CODE REVIEW

File: lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c [hunk 1]
```diff
@@ -297,6 +297,10 @@ int pthread_mutex_unlock( pthread_mutex_t * mutex )
 
     if( iStatus == 0 )
     {
+        /* Enter the critical section so that
+         * mutex is unlocked AND owner is updated atomically */
+        taskENTER_CRITICAL();
+
         /* Call the correct FreeRTOS mutex unlock function based on mutex type. */
         if( pxMutex->xAttr.iType == PTHREAD_MUTEX_RECURSIVE )
         {
```
Comment (gordonwang0) [inline lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c]: > FreeRTOS API functions must not be called from within a critical section. 

See: https://www.freertos.org/taskENTER_CRITICAL_taskEXIT_CRITICAL.html

Comment (lt72) [reply lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c]: Just move the critical section around the update of xTaskOwner.

File: lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c [hunk 2]
```diff
@@ -297,6 +297,10 @@ int pthread_mutex_unlock( pthread_mutex_t * mutex )
 
     if( iStatus == 0 )
     {
+        /* Suspend the sheduler so that
+         * mutex is unlocked AND owner is updated atomically */
+        vTaskSuspendAll();
```
Comment (huguesBouvier) [inline lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c]: Could you try to put a mutex here instead? suspendall is violent

Comment (lt72) [reply lib/FreeRTOS-Plus-POSIX/source/FreeRTOS_POSIX_pthread_mutex.c]: vTaskSuspendAll( ) is less invasive than disabling interrupts and also is the only other native way in FreeRTOS to use a critical section. Adding a mutex to guard the state would be more expensive in terms of RAM, so I would prefer to go ahead with this fix for now, and reconsider later, eventually, after applying the other fixes for POSIX layer.

**Pull request merged**